Backport from php-5 which will only use the dn_skipname function if the host 
system supports it (uClibc for example does not).

--- ext/standard/dns.c
+++ ext/standard/dns.c
@@ -275,6 +275,7 @@ PHP_FUNCTION(checkdnsrr)
 #define MAXHOSTNAMELEN  256
 #endif /* MAXHOSTNAMELEN */
 
+#if HAVE_DN_SKIPNAME
 /* {{{ proto int getmxrr(string hostname, array mxhosts [, array weight])
    Get MX records corresponding to a given Internet host name */
 PHP_FUNCTION(getmxrr)
@@ -355,6 +356,7 @@ PHP_FUNCTION(getmxrr)
 	RETURN_TRUE;
 }
 /* }}} */
+#endif /* HAVE_DN_SKIPNAME */
 
 #endif
 /*
--- ext/standard/basic_functions.c
+++ ext/standard/basic_functions.c
@@ -449,7 +449,9 @@ function_entry basic_functions[] = {
 
 #if HAVE_RES_SEARCH && !(defined(__BEOS__) || defined(PHP_WIN32) || defined(NETWARE))
 	PHP_FE(checkdnsrr,														NULL)
+# if HAVE_DN_SKIPNAME
 	PHP_FE(getmxrr,second_and_third_args_force_ref)
+# endif
 #endif
 
 	PHP_FE(getmyuid,														NULL)
